collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -337,19 +336,9 @@ public void granted() {
                     Collect.createODKDirs();
                     setupFields(savedInstanceState);
                     loadForm();
-
-                    /**
-                     * Since onResume is called after onCreate we check to see if
-                     * it was called without the permissions that are required. If so then
-                     * we call it.This is especially useful for cases where a user might revoke
-                     * permissions to storage and not know the implications it has on the form entry.
-                     */
-                    if (onResumeWasCalledWithoutPermissions) {
-                        onResume();
-                    }
+                    setup();
```

Comment:
Why is this called here? In normal execution, won't it just be called again in `onResume`?

Reply:
I haven't changed the original logic here. Let me check what is the impact of only calling it from `onResume`